core/trino-main/src/test/java/io/trino/connector/MockConnectorColumnHandle.java

```diff
@@ -69,7 +69,8 @@ public boolean equals(Object o)
             return false;
         }
         MockConnectorColumnHandle other = (MockConnectorColumnHandle) o;
-        return Objects.equals(name, other.name);
+        return Objects.equals(name, other.name) &&
+                Objects.equals(type, other.type);
```

Comment:
For some reason some ColumnHandles have partial equality
https://github.com/trinodb/trino/blob/02b59cd274f5314ef3c65ef0064b0371da449537/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcColumnHandle.java#L121-L127

i don't know why and i do hope this is not necessary there, but still worth noting

Reply:
in the context of connectors, I think it should be okay since column handle maps to a logical column in the datasource - so it should be uniquely identified based on name.

For `MockConnectorColumnHandle` at least, having this equality is good for asserting the expected plans accurately. e.g. https://github.com/trinodb/trino/pull/7288#discussion_r595727455